src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3545,18 +3545,21 @@ private int GetOrCreatePlaceholderSlot(object identifier, TypeWithAnnotations ty
 
             var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
             var placeholders = placeholdersBuilder.ToImmutableAndFree();
-            TypeSymbol? bestType = BestTypeInferrer.InferBestType(placeholders, walker._conversions, ref discardedUseSiteInfo);
+            TypeSymbol? bestType = BestTypeInferrer.InferBestType(placeholders, walker._conversions, ref discardedUseSiteInfo, unwrapFunctionType: false);
 
             TypeWithAnnotations inferredType;
             if (bestType is { })
             {
                 // Note: so long as we have a best type, we can proceed.
-                var bestTypeWithObliviousAnnotation = TypeWithAnnotations.Create(bestType);
+                var unwrappedType = bestType is FunctionTypeSymbol functionType ?
```

Comment:
>var unwrappedType = bestType is FunctionTypeSymbol functionType ? [](http://example.com/codeflow?start=16&length=65)

Would it be better to pass ```unwrapFunctionType: true``` above?

Reply:
Based on your suggestion offline, I've changed the various methods to return a `TypeSymbol` and a `bool` rather than requiring callers to pass in the correct value.